### PR TITLE
OpenBots.Core Cleanup

### PR DESCRIPTION
**Pull request by openbots-ch**
Cleaning up OpenBots.Core to minimize modifications needed during normal development. Moving exclusive functionality to components utilizing it.

**Leaving open for now.**

### CODE REVIEW

File: OpenBots.Commands/OpenBots.Commands.System/LaunchRemoteDesktopCommand.cs [hunk 1]
```diff
@@ -100,5 +115,18 @@ public override string GetDisplayValue()
         {
             return base.GetDisplayValue() + $" [Machine '{v_MachineName}']";
         }
+
+        public void LaunchRDPDisplayManager(object sender, EventArgs e)
+        {
+            frmDisplayManager displayManager = new frmDisplayManager();
+            displayManager.ShowDialog();
+            displayManager.Close();
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.System/LaunchRemoteDesktopCommand.cs:123]: You seem to be opening and closing this form without actually returning anything.

File: OpenBots.Studio/UI/Forms/frmSettings.cs [hunk 2]
```diff
@@ -399,91 +319,12 @@ private void btnSelectAttendedTaskFolder_Click(object sender, EventArgs e)
...
-                LocalTCPClient.StartListening(portNumber);
-            }
-        }
-
-        private void btnStopListening_Click(object sender, EventArgs e)
-        {
-            DisableListenerButtons();
-            LocalTCPClient.StopAutomationListener();
-        }
+        //private void btnLaunchDisplayManager_Click(object sender, EventArgs e)
```
Comment (openbots-ff) [inline OpenBots.Studio/UI/Forms/frmSettings.cs:322]: remove leftover comment?

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: OpenBots.Core/IO/JavaInterface.cs [hunk 3]
```diff
@@ -6,7 +6,7 @@ namespace OpenBots.Core.IO
 {
     public class JavaInterface
     {
-        private Process Create(string jarName, string args)
+        public Process Create(string jarName, string args)
         {
             var jarLibary = Path.Combine(Path.GetDirectoryName(Application.ExecutablePath), "Resources", jarName);
```
Comment (openbots-ff) [inline OpenBots.Core/IO/JavaInterface.cs:11]: I believe this path will point to the bin directory in the openbots.studio project, but the .jar resource you're using is located in the command resources. Is this working?

**Review by openbots-ff — CHANGES_REQUESTED** [review]

**Review by openbots-ff — APPROVED** [review]

File: OpenBots.Studio/UI/Forms/frmSettings.cs (hunk 2 above)
Comment (openbots-ch) [reply OpenBots.Studio/UI/Forms/frmSettings.cs:323]: done

File: OpenBots.Commands/OpenBots.Commands.System/LaunchRemoteDesktopCommand.cs (hunk 1 above)
Comment (openbots-ch) [reply OpenBots.Commands/OpenBots.Commands.System/LaunchRemoteDesktopCommand.cs:123]: Will be fixed in a second PR. The manager itself isn't particularly useful at the moment in general.

**Review by openbots-ff — APPROVED** [review]

File: OpenBots.Studio/Utilities/GlobalHook.cs [hunk 4]
```diff
@@ -458,8 +458,7 @@ private static void BuildMouseCommand(IntPtr lParam, MouseMessages mouseMessage)
                 if (mouseEventClickType != "None")
                 {
                     IntPtr winHandle = WindowFromPoint(hookStruct.Pt);
-
-                    int length = GetWindowText(winHandle, _buffer, _buffer.Capacity);
+                    _ = GetWindowText(winHandle, _buffer, _buffer.Capacity);
```
Comment (openbots-ab) [inline OpenBots.Studio/Utilities/GlobalHook.cs:461]: Can you please check this?

**Review by openbots-ab — CHANGES_REQUESTED** [review]
I left a comment for you, please check.

File: OpenBots.Core/IO/JavaInterface.cs (hunk 3 above)
Comment (openbots-ch) [reply OpenBots.Core/IO/JavaInterface.cs:11]: We can come back on this. The command isnt working on 1.0.4 either, and we should probably eliminate the jar method altogether.

File: OpenBots.Studio/Utilities/GlobalHook.cs (hunk 4 above)
Comment (openbots-ch) [reply OpenBots.Studio/Utilities/GlobalHook.cs:461]: Just replaced the length variable with a discard since we weren't using the created length variable.

**Review by openbots-ab — APPROVED** [review]

**Pull request merged**